packages/@lwc/engine-dom/src/patches/global-registry.ts

```diff
@@ -0,0 +1,398 @@
+            attributeChangedCallback!.call(instance, name, null, newValue);
+        }
+    });
+}
+
+function getDefinitionForConstructor(constructor: CustomElementConstructor): Definition {
+    if (!constructor || !constructor.prototype || typeof constructor.prototype !== 'object') {
+        throw new TypeError(`The referenced constructor is not a constructor.`);
+    }
+    let definition = definitionForConstructor.get(constructor);
+    if (isUndefined(definition)) {
+        definition = createDefinitionRecord(constructor);
+        definitionForConstructor.set(constructor, definition);
+    }
+    return definition;
+}
+
+export function patchCustomElementRegistry() {
+    const { customElements: nativeRegistry } = window;
+    const { define: nativeDefine, whenDefined: nativeWhenDefined, get: nativeGet } = nativeRegistry;
+
+    if ((nativeRegistry as any)[registryPatchedSymbol]) {
+        throw new Error(
+            'patchCustomElementRegistry was called twice in the same JavaScript environment. ' +
+                'Please ensure you are loading the LWC engine only once. ' +
+                'If this is a Jest environment, check your Jest/JSDOM configuration.'
+        );
+    }
+    (nativeRegistry as any)[registryPatchedSymbol] = true;
+
+    // patch for the global registry define mechanism
+    CustomElementRegistry.prototype.define = function define(
+        this: CustomElementRegistry,
+        tagName: string,
+        constructor: CustomElementConstructor,
+        options?: ElementDefinitionOptions
+    ): void {
+        if (options && options.extends) {
+            throw new DOMException(
+                'NotSupportedError: "extends" key in customElements.define() options is not supported.'
+            );
+        }
+        nativeGet.call(this, tagName); // SyntaxError if The provided name is not a valid custom element name.
+        tagName = tagName.toLowerCase();
+        if (!isUndefined(definitionsByTag.get(tagName))) {
+            throw new DOMException(
+                `Failed to execute 'define' on 'CustomElementRegistry': the name "${tagName}" has already been used with this registry`
+            );
+        }
+        const definition = getDefinitionForConstructor(constructor);
+        if (!isUndefined(definitionsByClass.get(constructor))) {
+            throw new DOMException(
+                `Failed to execute 'define' on 'CustomElementRegistry': this constructor has already been used with this registry`
+            );
+        }
+        definitionsByTag.set(tagName, definition);
+        definitionsByClass.set(constructor, definition);
+        let PivotCtor = pivotCtorByTag.get(tagName);
+        if (isUndefined(PivotCtor)) {
+            PivotCtor = createPivotingClass(tagName, definition);
+            // Register a pivoting class which will handle global registry initializations
+            nativeDefine.call(nativeRegistry, tagName, PivotCtor);
+        }
```

Comment:
@caridy AFAICT this `if` block will always be called, because the `PivotCtor` will always exist. I cannot see a case where we get this far into the `define()` function and yet the `PivotCtor` has already been defined for this `tagName`. If it was defined, then we wound have thrown the error for the duplicate `tagName`.